migration.py

```diff
@@ -23,21 +24,27 @@
 # NOTE: this is parsed in the main function
 TFE_VCS_CONNECTION_MAP = None
 SENSITIVE_DATA_MAP = None
+SOURCE_WORKSPACE_LIST = []
 
 
-def main(migrator, delete_all, no_confirmation, migrate_all_state, migrate_sensitive_data, tfe_verify_source):
+def main(migrator, delete_all, no_confirmation, migrate_all_state, migrate_sensitive_data, migrate_select_workspaces, tfe_verify_source):
 
     if delete_all:
         migrator.delete_all_from_target(no_confirmation)
     elif migrate_sensitive_data:
         migrator.migrate_sensitive()
     else:
-        migrator.migrate_all(migrate_all_state, tfe_verify_source)
+        migrator.migrate_all(migrate_all_state, migrate_select_workspaces, tfe_verify_source)
```

Comment:
I would not overload the `migrate_all` function, and instead, add another function and additional control logic to support this workflow.